testhelpers/docker_registry.go

```diff
@@ -88,6 +103,36 @@ func ReadOnly(handler http.Handler) http.Handler {
 	})
 }
 
+func Custom(handler http.Handler, permissions map[string]ImagePrivileges) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if strings.Contains(r.URL.Path, "blobs") ||
+			strings.Contains(r.URL.Path, "manifests") ||
+			strings.Contains(r.URL.Path, "tags") {
+			elems := strings.Split(r.URL.Path, "/")
+			elems = elems[1:]
+			name := elems[len(elems)-1]
```

Comment:
A helper function might be useful here (to make the code more readable)

Reply:
I'm having trouble here. `name` is supposed to be the image tag, correct? The same value passed into the `permissions` mapping?

The following is what I'm seeing for `r.URL.Path` with `remote.NewImage()`. Am I using it right?

```shell
# image tag: 192.168.1.164:63737/aemengo/test
Path: /v2/aemengo/test/blobs/sha256:60df352a35c779a36da39e670ddc90742f64209045365ea0ac4f5eb83422a615
Path: /v2/aemengo/test/blobs/uploads/
Path: /v2/aemengo/test/blobs/uploads/4831389563158288344
Path: /v2/aemengo/test/blobs/uploads/4831389563158288344
Path: /v2/aemengo/test/manifests/latest
Path: /v2/aemengo/test/manifests/latest
Path: /v2/aemengo/test/blobs/uploads/
Path: /v2/aemengo/test/blobs/uploads/261049867304784443
```

Reply:
well, in this case, `name` should be `test` and that should be the key to find in `permissions`. that you put before it is not going to be used to determine the permission (at least from the test perspective)

Reply:
Hmm... this playground is giving me a blank name: https://play.golang.org/p/Zn2EqfiHGb5

I think we can't assume that `test` will be at the end of the path...

Reply:
Yeap, it's my mistake I will fix it ;) thanks @natalieparellano and @aemengo

Reply:
@aemengo , @natalieparellano 

What about something like [this](https://play.golang.org/p/QfbivVx2xIt) to parse the URL and extract the `name`, @aemengo I am trying to handle you mentioned 😄 .

I like the regex to isolate the string to V2/[whatever]/(blobs|manifest|tag), but I open to suggestion to get [whatever] in a optimal way

Reply:
@jjbustamante Why so complicated? 😅 Take a look at [this one](https://play.golang.org/p/F3b-dSljAQT)

Reply:
hahaha.. come on! I hate regex, thanks! any thoughts on best option to remove the /V2/****/(blobs|manifest|tag) stuff? and just keep the name?

Reply:
If you use the `FindStringSubmatch` method, the name will be 2nd value in an array of 2 elements. 👍🏾 

```
matches := s.FindStringSubmatch(str)
matches[1] //<--- name
```

Reply:
Awesome! I like it! thanks

Reply:
I fixed the issue and took into consideration the suggestion, the logic was moved to the helper function and I used the regex pattern matching to find the `name` see [commit](https://github.com/buildpacks/imgutil/pull/125/commits/4261e2896afc176ea6a4967a4bd255f780340892)